tmt/utils.py

```diff
@@ -4779,6 +4780,14 @@ def git_clone(
 
     if can_change:
         url = clonable_git_url(url)
+
+    # Add env GIT_HTTP_CONNECT_TIMEOUT to avoid being stuck when running 'git clone'
+    timeout = None if env is None else env.get('GIT_HTTP_CONNECT_TIMEOUT', None)
+    if timeout is None:
+        if env is None:
+            env = {}
+        env['GIT_HTTP_CONNECT_TIMEOUT'] = '5'
```

Comment:
This looks weird.

1. is the variable documented anywhere? I can't find any official source documenting `GIT_HTTP_CONNECT_TIMEOUT`. It appears in https://github.com/teemtee/tmt/issues/2317#issue-1887413090 but where does it come from?
2. tmt should probably have better control over this behavior, i.e. there should be a `TMT_GIT_HTTP_TIMEOUT` envvar (we already do have `TMT_GIT_CREDENTIALS_*`, therefore `TMT_GIT_*` namespace seems to be fitting).
3. why is there `retry_times` but no `timeout`? In other words, why one parameter of "retry cloning" is passed via a function parameter, and the second one hidden among environment variables? The function can and should probably treat both parameters in the same way, there should be envvar for each of them:.
   ```python
   def git_clone(..., retries: Optional[int] = None, timeout: Optional[None] = None):
       retries = retries or os.environ.get('TMT_GIT_CLONE_RETRIES', DEFAULT_GIT_CLONE_RETRIES)
       timeout = timeout or os.environ.get('TMT_GIT_CLONE_TIMEOUT', DEFAULT_GIT_CLONE_TIMEOUT)
   ```
   Then, these values would be used to spike `env` with git-specific variables or handle the countdown.

Reply:
> 1. is the variable documented anywhere? I can't find any official source documenting GIT_HTTP_CONNECT_TIMEOUT. It appears in https://github.com/teemtee/tmt/issues/2317#issue-1887413090 but where does it come from?

It was just copied from the issue #2317 actually as I failed to find out what environment variable can be used to control the timeout of `git http connection` from the official documents of gitlab.com.

Since we don't have any environment var to set the timeout of command `git clone ...`, can we use `timeout` to run a command with a time limit? (Command `timeout` is from package `coreutils`)  e.g.

```bash
huanli@FOO:tmp$ timeout 2 git clone https://github.com/teemtee/tmt.git /tmp/tmt01
Cloning into '/tmp/tmt01'...
remote: Enumerating objects: 19081, done.
huanli@FOO:tmp$ :  93% (3397/3652)
huanli@FOO:tmp$ echo $?
124
huanli@FOO:tmp$ 
huanli@FOO:tmp$ timeout 600 git clone https://github.com/teemtee/tmt.git /tmp/tmt02
Cloning into '/tmp/tmt02'...
remote: Enumerating objects: 19081, done.
remote: Counting objects: 100% (3652/3652), done.
remote: Compressing objects: 100% (600/600), done.
remote: Total 19081 (delta 3285), reused 3140 (delta 3050), pack-reused 15429
Receiving objects: 100% (19081/19081), 4.70 MiB | 1.82 MiB/s, done.
Resolving deltas: 100% (13002/13002), done.
huanli@FOO:tmp$ echo $?
0
```

Reply:
`Common.run()` does have a parameter `timeout`, which should be exactly what we're looking for.

Reply:
Updated.